src/routes/accounts.py

```
@@ -20,6 +20,247 @@
+    jwt_manager: JWTAuthManagerInterface = Depends(get_jwt_auth_manager),
+):
+    user_result = await db.execute(select(UserModel).where(UserModel.email == request.email))
+    db_user = user_result.scalars().first()
+
+    if not db_user or not db_user.verify_password(request.password):
+        raise HTTPException(
+            status_code=status.HTTP_401_UNAUTHORIZED,
+            detail="Invalid email or password."
+        )
+
+    if not db_user.is_active:
+        raise HTTPException(
+            status_code=status.HTTP_403_FORBIDDEN,
+            detail="User account is not activated."
+        )
+
+    try:
+        token_data = {"user_id": db_user.id, "email": db_user.email}
+
+        access_token = jwt_manager.create_access_token(data=token_data)
+        refresh_token = jwt_manager.create_refresh_token(data=token_data)
+
+        refresh_token_obj = RefreshTokenModel(token=refresh_token, user_id=db_user.id)
+        db.add(refresh_token_obj)
+
+        await db.commit()
+
+        return UserLoginResponseSchema(
+            access_token=access_token,
+            refresh_token=refresh_token
+        )
+
+    except SQLAlchemyError:
+        await db.rollback()
+        raise HTTPException(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            detail="An error occurred while processing the request."
+        )
+
+
+@router.post("/refresh/", response_model=TokenRefreshResponseSchema)
+async def refresh_access_token(
+    request: TokenRefreshRequestSchema,
+    db: AsyncSession = Depends(get_db),
+    jwt_manager: JWTAuthManagerInterface = Depends(get_jwt_auth_manager),
+):
+    try:
+        token_data = jwt_manager.decode_refresh_token(request.refresh_token)
+        user_id = token_data.get("user_id")
+        if not user_id:
+            raise HTTPException(
+                status_code=status.HTTP_400_BAD_REQUEST,
+                detail="Invalid token payload."
+            )
+    except Exception as e:
```

Catching a generic `Exception` is too broad and can hide underlying issues. It's better to catch specific exceptions that the `jwt_manager` might raise (e.g., subclasses of `BaseSecurityError` or specific JWT-related errors) to handle them appropriately and avoid suppressing unexpected errors.